rmw_zenoh_cpp/src/detail/liveliness_utils.cpp

```diff
@@ -161,13 +173,34 @@ std::vector<std::string> split_keyexpr(
 std::string qos_to_keyexpr(rmw_qos_profile_t qos)
 {
   std::string keyexpr = "";
+  // Reliability.
   keyexpr += std::to_string(qos.reliability);
   keyexpr += QOS_DELIMITER;
+  // Durability.
   keyexpr += std::to_string(qos.durability);
   keyexpr += QOS_DELIMITER;
+  // History.
   keyexpr += std::to_string(qos.history);
-  keyexpr += QOS_HISTORY_DELIMITER;
+  keyexpr += QOS_COMPONENT_DELIMITER;
   keyexpr += std::to_string(qos.depth);
+  // Deadline.
+  keyexpr += QOS_DELIMITER;
```

Comment:
This is going to seem really pedantic, but I'd prefer if this is moved up two lines.  The reason is so that the delimiter is always attached to the end of the thing it is delimiting.  At the end of the day, it won't matter too much, but I find it easier to read:

```suggestion
  keyexpr += std::to_string(qos.depth);
  keyexpr += QOS_DELIMITER;
  // Deadline.
```

(alternatively, you could leave this QOS_DELIMITER where it is, but then change the Reliability, Durability, and History QOS_DELIMITER to be in the one below it.  Either is fine, I just want it to be consistent one way or the other)

Reply:
Totally valid feedback. Done https://github.com/ros2/rmw_zenoh/pull/210/commits/17f841924d1ccf092d97c55809e26202c9b436b6